src/widgets/combobox.ts

```diff
@@ -3,29 +3,55 @@ import { ElementArrayFinder } from "../wdio";
 
 
 export class ComboBox extends Widget {
-
     public async click(): Promise<void> {
-        return this.elem.click();
+        return this.elem.byCSS('dropdown-toggle').click();
+    }
+
+    public async isOptionsListOpen(): Promise<boolean> {
+        const nOptions = await this.allByCSS('.ag-cell-value').count();
+        return nOptions > 0;
+    }
+
+    public async openOptionsList(): Promise<void> {
+        const isOpen = await this.isOptionsListOpen();
+        if (!isOpen) {
+            await this.elem.click();
+            await this.waitUntil(async () => this.isOptionsListOpen());
+        }
+    }
+
+    public async closeOptionsList(): Promise<void> {
+        const isOpen = await this.isOptionsListOpen();
+        if (isOpen) {
+            await this.elem.click();
+            await this.waitUntil(async () => !(await this.isOptionsListOpen()));
+        }
     }
 
     public async getOptions(): Promise<string[]> {
+        await this.openOptionsList();
+
         let content: string[] = [];
         let rows: ElementArrayFinder = this.allByCSS('.ag-cell-value');
         let numberOfItems: number = await rows.count();
         for (let i = 0; i < numberOfItems; i++) {
             let text: string = await rows.get(i).getText();
             content.push(text);
         }
-        return content;
-    }
 
-    public async selectOptionByNumber(i: number): Promise<void> {
-        return this.allByCSS(`[role='row'][row-index='` + i + `']`).get(1).click();
+        await this.closeOptionsList();
+
+        return content;
     }
 
     public async selectOptionByText(text: string): Promise<void> {
         const options = await this.getOptions();
         const optionIndex = options.findIndex((option) => option === text);
         return this.selectOptionByNumber(optionIndex);
     }
+
+    public async selectOptionByNumber(i: number): Promise<void> {
+        await this.openOptionsList();
+        await this.allByCSS(`[role='row'][row-index='` + i + `']`).get(1).click();
```

Comment:
I am just curious in why here it is needed to get the 1nth and not the 0nth

Reply:
I think it's becase position 0 is the placeholder (?)

Reply:
Not sure because this code was already there and hasn't been changed, but I think that there are 2 items on each row/option. Thus, I guess that the graphical item for each option goes into position 2.